### PR TITLE
kv for saplibrary

### PR DESCRIPTION
**Pull request by YunziZhangAzure**
## Problem
<Please describe what problem this PR is trying to resolve>
According to the requirement of KeyVault, #773
Two kvs should be created within the rg of saplibrary.
username/password of downloader should be stored in the user accessible kv.

## Solution
<Please elaborate the solution for the problem>
1. Create two KVs within saplibrary group
2. Store username/password of downloader in the user kv.
3. The test pipeline for saplib is also modified.

## Tests
<Please provide steps to test the PR>
input json is as follows:

{
    "infrastructure": {
        "region": "eastus",
        "environment": "global",
        "resource_group": {
            "is_existing": "false",
            "arm_id": ""
        },
        "deployer": {
            "resource_group": {
                "name": "" <- the rg name of deployer
            },
            "msi": {
                "name": "" <- the name of msi of deployer
            },
            "users": {
                "object_id": []   <- populate the object_id of users to give these users the access to the secrets when using Azure portal
            }
        }
    },
    "software": {
        "downloader": {
            "credentials": {
                "sap_user": "",
                "sap_password": ""
            }
        }
    }
}

## Notes
<Additional comments for the PR>
new test pipeline for saplib is:
https://dev.azure.com/azuresaphana/Azure-SAP-HANA/_build?definitionId=114

### CODE REVIEW

**Review by rukawata — COMMENTED** [review]
Overall looks good! Some small comments :)

File: deploy/terraform/bootstrap/sap_library/saplibrary.json [hunk 1]
```diff
@@ -5,26 +5,25 @@
...
-        "is_existing": "false",
-        "arm_id": "",
-        "file_share": {
-            "enable_deployment": true,
-            "name": "sapbits"
         },
-        "sapbits_blob_container": {
-            "enable_deployment": true,
-            "name": "sapbits"
+        "deployer": {
```
[the remark this answers is not in the document]
Comment (YunziZhangAzure) [reply deploy/terraform/bootstrap/sap_library/saplibrary.json:9]: Will do.

File: .pipeline/templates/saplib/create-saplib-steps.yml [hunk 2]
```diff
@@ -38,6 +39,10 @@ steps:
       cp ${repo_dir}/deploy/terraform/bootstrap/sap_library/saplibrary.json ${ws_dir}/saplibrary.json
       cat ${ws_dir}/saplibrary.json \
       | jq --arg environment "${saplib_prefix}" .infrastructure.environment\ =\ \$environment \
+      | jq --arg deployer_rg_name "UNIT-EAUS-MGMT-INFRASTRUCTURE" .infrastructure.deployer.resource_group.name\ =\ \$deployer_rg_name \
+      | jq --arg msi_name "UNIT-EAUS-MGMT-msi" .infrastructure.deployer.msi.name\ =\ \$msi_name \
+      | jq --arg downloader_username "$(hana-smp-nancyc-username)" .software.downloader.credentials.sap_user\ =\ \$downloader_username \
+      | jq --arg downloader_password "$(hana-smp-nancyc-password)" .software.downloader.credentials.sap_password\ =\ \$downloader_password \
```
Comment (rukawata) [inline .pipeline/templates/saplib/create-saplib-steps.yml:45]: This needs to be changed after the change in JSON :)

**Review by rukawata — APPROVED** [review]
LGTM!

**Pull request merged**